### PR TITLE
FEATURE: Automate release of surround from Codefresh

### PR DESCRIPTION
**Pull request by AkshatBajaj**

### CODE REVIEW

**Comment by ucokzeko** [comment]
As discussed on the call:
1. Look into automatically pushing tags to repo before releasing to `Pypi`
2. Look into linking test Codefresh pipeline into the release pipeline

**Comment by AkshatBajaj** [comment]
> As discussed on the call:
> 
> 1. Look into automatically pushing tags to repo before releasing to `Pypi`
> 2. Look into linking test Codefresh pipeline into the release pipeline

I used a different approach now, it is again based on the tags, please have a look

File: codefresh.yml [hunk 1]
```diff
@@ -68,29 +68,63 @@ steps:
...
-
-  #     # Upload package for distribution
-  #     - python3 -m twine upload --repository pypi dist/*
-  #   when:
-  #     branch:
-  #       only:
-  #         - /([0-9]+)\.([0-9]+)\.([0-9]+)?$/
-  #   stage: "release"
+  release:
+    type: parallel
```
Comment (ucokzeko) [inline codefresh.yml:72]: Add `stage: "release"` above this

File: codefresh.yml [hunk 2]
```diff
@@ -68,29 +68,63 @@ steps:
...
+          - python3 setup.py sdist bdist_wheel
+
+          # Upload package for distribution
+          - python3 -m twine upload --repository pypi dist/*
+        when:
+            condition:
+              all:
+                isMainRepo: '"${{CF_REPO_OWNER}}" == "a2i2"'
+                isReleaseBranch: 'match("${{CF_BRANCH}}", "/surround-v([0-9]+)\.([0-9]+)\.([0-9]+)?$", false) == true'
+        stage: "release"
```
Comment (ucokzeko) [inline codefresh.yml:101]: This can be removed

File: codefresh.yml [hunk 3]
```diff
@@ -68,29 +68,63 @@ steps:
...
+          - python3 setup.py sdist bdist_wheel
+
+          # Upload package for distribution
+          - python3 -m twine upload --repository pypi dist/*
+        when:
+            condition:
+              all:
+                isMainRepo: '"${{CF_REPO_OWNER}}" == "a2i2"'
+                isReleaseBranch: 'match("${{CF_BRANCH}}", "/surround-cli-v([0-9]+)\.([0-9]+)\.([0-9]+)?$", false) == true'
+        stage: "release"
```
Comment (ucokzeko) [inline codefresh.yml:130]: This too

**Review by ucokzeko — CHANGES_REQUESTED** [review]
Please do the same for `test` stage as well

What's the reason we are going back to version tagging for a release?

**Comment by AkshatBajaj** [comment]
> Please do the same for `test` stage as well
> 
> What's the reason we are going back to version tagging for a release?

It is based on Rohan's suggestion, he recommended we are better off with the tags and it is possible with the Codefresh for monorepos, we anyway solved the problem we were facing, so seems like a good approach to me :)

**Comment by ucokzeko** [comment]
> It is based on Rohan's suggestion, he recommended we are better off with the tags and it is possible with the Codefresh for monorepos, we anyway solved the problem we were facing, so seems like a good approach to me :)

Yeah, it's all good :+1:
Just checking the reasoning behind this decision.

**Review by ucokzeko — APPROVED** [review]

**Pull request merged**